isdproject/src/Profile.js

```diff
@@ -170,6 +172,38 @@ const Profile = () => {
     }
   };
 
+  const [orderIdSearch, setOrderIdSearch] = useState('');
+const [dateSearch, setDateSearch] = useState('');
+
+const handleSearchOrderHistory = () => {
```

Comment:
This search has to be done via server-side search, not a front-end filter. Here are some docs to help you get started building a query: https://www.mongodb.com/docs/manual/reference/method/db.collection.find/